### ISSUE TITLE
Haskell package pure-zlib is missing test dependencies.

### ISSUE THREAD

**Issue by TikhonJelvis** [NixOS/nixpkgs#37499]
## Issue description

The `pure-zlib` package fails to build because of missing test dependencies:

```
building '/nix/store/9d3l4z282i4yp8fw8mz36dkls2ilpirv-pure-zlib-0.6.drv'...
setupCompilerEnvironmentPhase
Build with /nix/store/z6jgjdggj7xkrm8l8bd5xn7pq6cw85wy-ghc-8.2.2.
unpacking sources
unpacking source archive /nix/store/x38aw9sgg4c1d02fzdngalyd2if2wznb-pure-zlib-0.6.tar.gz
source root is pure-zlib-0.6
setting SOURCE_DATE_EPOCH to timestamp 1480113731 of file pure-zlib-0.6/test/Test.hs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/private/var/folders/my/71d693qj5szdm986z1wj06krlrw5ms/T/nix-build-pure-zlib-0.6.drv-0/package.conf.d -j8 -threaded
[1 of 1] Compiling Main             ( Setup.hs, /private/var/folders/my/71d693qj5szdm986z1wj06krlrw5ms/T/nix-build-pure-zlib-0.6.drv-0/Main.o )
Linking Setup ...
clang-5.0: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
clang-5.0: warning: argument unused during compilation: '-nopie' [-Wunused-command-line-argument]
configuring
configureFlags: --verbose --prefix=/nix/store/va5ahfffmkr582lxf1smzx1gwxd9y0zs-pure-zlib-0.6 --libdir=$prefix/lib/$compiler --libsubdir=$pkgid --docdir=/nix/store/cyyl0p5z9b4498h0nd4wknwffznn2356-pure-zlib-0.6-doc/share/doc --with-gcc=clang --package-db=/private/var/folders/my/71d693qj5szdm986z1wj06krlrw5ms/T/nix-build-pure-zlib-0.6.drv-0/package.conf.d --ghc-option=-optl=-Wl,-headerpad_max_install_names --ghc-option=-j8 --disable-split-objs --disable-library-profiling --disable-profiling --enable-shared --disable-coverage --enable-library-vanilla --enable-executable-dynamic --enable-tests --extra-include-dirs=/nix/store/j6ifwrr83k1c5nmjfx1p63nydpqsbqd6-libc++-5.0.1/include --extra-lib-dirs=/nix/store/j6ifwrr83k1c5nmjfx1p63nydpqsbqd6-libc++-5.0.1/lib --extra-lib-dirs=/nix/store/a67rhbgwb41j11j6mgww3y0gq4pg0lmp-ncurses-6.0-20171125/lib --extra-lib-dirs=/nix/store/lasa9z5jcwhq3djd3mw8q9b0a75da5as-gmp-6.1.2/lib --extra-include-dirs=/nix/store/5hymrhfv7zm20ackp3i3di2n7amnlvva-libiconv-osx-10.11.6/include --extra-lib-dirs=/nix/store/5hymrhfv7zm20ackp3i3di2n7amnlvva-libiconv-osx-10.11.6/lib --extra-include-dirs=/nix/store/5hymrhfv7zm20ackp3i3di2n7amnlvva-libiconv-osx-10.11.6/include --extra-lib-dirs=/nix/store/5hymrhfv7zm20ackp3i3di2n7amnlvva-libiconv-osx-10.11.6/lib
Configuring pure-zlib-0.6...
CallStack (from HasCallStack):
  die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:948:20 in Cabal-2.0.1.0:Distribution.Simple.Configure
  configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:470:12 in Cabal-2.0.1.0:Distribution.Simple.Configure
  configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:570:20 in Cabal-2.0.1.0:Distribution.Simple
  confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:67:5 in Cabal-2.0.1.0:Distribution.Simple.UserHooks
  configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:174:19 in Cabal-2.0.1.0:Distribution.Simple
  defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:119:27 in Cabal-2.0.1.0:Distribution.Simple
  defaultMain, called at Setup.hs:2:8 in main:Main
Setup: Encountered missing dependencies:
HUnit >=1.2 && <1.4, QuickCheck >=2.7 && <2.9
builder for '/nix/store/9d3l4z282i4yp8fw8mz36dkls2ilpirv-pure-zlib-0.6.drv' failed with exit code 1
error: build of '/nix/store/9d3l4z282i4yp8fw8mz36dkls2ilpirv-pure-zlib-0.6.drv' failed
```

### Steps to reproduce

```
nix-shell -p pkgs.haskellPackages.pure-zlib
```

## Technical details

 - system: `"x86_64-darwin"`
 - host os: `Darwin 16.7.0, macOS 10.12.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.12pre5788_e3013543`
 - channels(z0028sn): `"nixpkgs-18.09pre132003.13e74a838db"`
 - nixpkgs: `/Users/z0028sn/.nix-defexpr/channels/nixpkgs`

**Comment by mpickering** [comment]
I jailbroke it and the the tests fail anyway.

```
Building executable 'deflate' for pure-zlib-0.6..
[1 of 1] Compiling Main             ( Deflate.hs, dist/build/deflate/deflate-tmp/Main.dyn_o )
Linking dist/build/deflate/deflate ...
running tests
Running 1 test suites...
Test suite test-zlib: RUNNING...
test-zlib: test/test-cases/randtest1.z: openBinaryFile: does not exist (No such file or directory)
Test suite test-zlib: FAIL
Test suite logged to: dist/test/pure-zlib-0.6-test-zlib.log
0 of 1 test suites (0 of 1 test cases) passed.
```

So it should be set to `dontCheck`. `nix-shell -p "haskell.lib.dontCheck (haskellPackages.pure-zlib)"` succeeds.

**Comment by peti** [comment]
The problem is that the release tarball on Hackage is incomplete, i.e. the required file `test/test-cases/randtest1.z` appears to be missing. This should be reported upstream (https://github.com/GaloisInc/pure-zlib) so that they can release a fixed version.

Ping: @acw

**Comment by acw** [comment]
This should be fixed in 0.6.1, which I just uploaded to Hackage. Please tell me (probably on a pure-zlib ticket?) if you have any further problems.
